pkg/utils/feature.go

```diff
@@ -0,0 +1,12 @@
+package utils
+
+var ClusterLegacyIngressLabel = "ext-managed.openshift.io/legacy-ingress-support"
+
+func HasUserManagedIngressFeature(labels map[string]string) bool {
+	if labels == nil {
+		return false
+	}
+	// Not having the label is assumed to mean that the deployment is a legacy cluster
+	legacyIngressLabel, labelExists := labels[ClusterLegacyIngressLabel]
+	return labelExists && legacyIngressLabel == "false"
```

Comment:
> I assume with this intention, we want to mark all clusters with `ext-managed.openshift.io/legacy-ingress-support=false` except the ones we want the CIO to progress in.

Thanks for calling this out @boranx. Implicitly, all clusters right now have 'legacy' ingress and do not support the new flow (`legacy-ingress-support=true`). 

In terms of written language, it comes out to `all clusters support legacy ingress unless otherwise specified`.

It is a little clunky, but it means that we don't have to label existing clusters for behavior they don't need to know anything about, until such time that it rolls out to production. Right now it means that we can switch on the feature by just labeling the `PublishingStrategy` CR with `legacy-ingress-support: false`, and test the behavior in isolation.